theano/tensor/tests/test_opt.py

```diff
@@ -1576,6 +1576,30 @@ def test_log_add():
     #TODO: (write and) test that the optimization works with Sum in addition to working with Add.
 
 
+def test_local_useless_slice():
+    x = tensor.matrix('x')
+    mode = compile.get_default_mode().including("local_useless_inc_subtensor")
+    # test with and without the useless slice
+    o1 = 2 * x[0, :]
+    f1 = theano.function([x], o1)
+    o2 = 2 * x[0]
+    f2 = theano.function([x], o2)
+    test_inp = numpy.random.randint(-10, 10, (4, 4)).astype('float32')
+    assert all(f1(test_inp)  == f1(test_inp)), "Before and after slice not the same" 
+    # test to see if slice is truely gone
+    apply_node = f1.maker.fgraph.toposort()[0]
+    subtens = apply_node.op
+    assert not any(isinstance(idx, slice) for idx in subtens.idx_list)
```

Comment:
Comparing two functions is the right idea, but rather than giving them two different graphs, you should compile one with the normal mode and one with `get_default_mode().excluding('local_useless_slice')` from the same graph.

Then you compare the output of the two functions and they should be equal. Make sure to compare the shape of the input also.